### ISSUE TITLE
linkerd check wrong viz tap certificate secret name

### ISSUE THREAD

**Issue by rootik** [linkerd/linkerd2#9628]
### What is the issue?

`linkerd check` reports missing tap API server cert, but the secret name it's looking for is wrong.
It seems `linkerd check` is looking for `linkerd-tap-tls` secret but the actual secret name is `tap-k8s-tls`.

All viz components are up and running healthy.

[This change](https://linkerd.io/2.12/tasks/upgrade/#post-upgrade-cleanup) might be related.

### How can it be reproduced?

Follow this [guide](https://linkerd.io/2.12/tasks/automatically-rotating-webhook-tls-credentials/)
and install the latest (30.3.3) linkerd-viz helm chart from [Linkerd repo](https://helm.linkerd.io/stable)
with the following values:
```
tap:
  externalSecret: true
  caBundle: |
    <PEM certificate>
tapInjector:
  externalSecret: true
  caBundle: |
    <PEM certificate>
```

run `linkerd check`

### Logs, error output, etc

```
linkerd-viz
-----------
× tap API server has valid cert
    secrets "linkerd-tap-tls" not found
    see https://linkerd.io/2.12/checks/#l5d-tap-cert-valid for hints
```
```
linkerd viz dashboard
Cannot connect to Linkerd Viz: secrets "linkerd-tap-tls" not found
Validate the install with: linkerd viz check
```

### output of `linkerd check -o short`

```
Linkerd core checks
===================

linkerd-identity
----------------
‼ issuer cert is valid for at least 60 days
    issuer certificate will expire on 2022-10-17T23:34:54Z
    see https://linkerd.io/2.12/checks/#l5d-identity-issuer-cert-not-expiring-soon for hints

linkerd-webhooks-and-apisvc-tls
-------------------------------
‼ proxy-injector cert is valid for at least 60 days
    certificate will expire on 2022-10-18T13:48:32Z
    see https://linkerd.io/2.12/checks/#l5d-proxy-injector-webhook-cert-not-expiring-soon for hints
‼ sp-validator cert is valid for at least 60 days
    certificate will expire on 2022-10-18T13:48:32Z
    see https://linkerd.io/2.12/checks/#l5d-sp-validator-webhook-cert-not-expiring-soon for hints
‼ policy-validator cert is valid for at least 60 days
    certificate will expire on 2022-10-18T13:48:32Z
    see https://linkerd.io/2.12/checks/#l5d-policy-validator-webhook-cert-not-expiring-soon for hints

Linkerd extensions checks
=========================

linkerd-viz
-----------
× tap API server has valid cert
    secrets "linkerd-tap-tls" not found
    see https://linkerd.io/2.12/checks/#l5d-tap-cert-valid for hints

Status check results are ×
```

### Environment

- Kubernetes version: 1.23.5
- Cluster environment: AKS
- Host OS: Ubuntu 18.04
- Linkerd version: stable-2.12.1
- linkerd-viz helm chart version: 30.3.3
- cert-manager helm chart version: 0.8.5
- cert-manager version: 1.9.1

### Additional context

linkerd and linkerd-viz related cert-manager resources:
```
# https://linkerd.io/2.12/tasks/automatically-rotating-control-plane-tls-credentials/
# https://linkerd.io/2.12/tasks/automatically-rotating-webhook-tls-credentials/
---
apiVersion: spv.no/v1
kind: AzureKeyVaultSecret
metadata:
  name: linkerd-ca-cert
  namespace: linkerd
spec:
  output:
    secret:
      #chainOrder: ensureserverfirst
      name: linkerd-ca-cert
      type: kubernetes.io/tls
  vault:
    name: <snip>
    object:
      name: linkerd-ca-cert
      type: secret
---
apiVersion: spv.no/v1
kind: AzureKeyVaultSecret
metadata:
  name: linkerd-webhook-ca-cert
  namespace: linkerd
spec:
  output:
    secret:
      #chainOrder: ensureserverfirst
      name: linkerd-webhook-ca-cert
      type: kubernetes.io/tls
  vault:
    name: <snip>
    object:
      name: linkerd-webhook-ca-{{ .Values.region }}-cert
      type: secret
---
apiVersion: cert-manager.io/v1
kind: Issuer
metadata:
  name: linkerd-trust-anchor
  namespace: linkerd
spec:
  ca:
    secretName: linkerd-ca-cert
---
apiVersion: cert-manager.io/v1
kind: Issuer
metadata:
  name: webhook-issuer
  namespace: linkerd
spec:
  ca:
    secretName: linkerd-webhook-ca-cert
---
# ignore if not using the viz extension
apiVersion: cert-manager.io/v1
kind: Issuer
metadata:
  name: webhook-issuer
  namespace: linkerd-viz
spec:
  ca:
    secretName: linkerd-webhook-ca-cert
---
apiVersion: cert-manager.io/v1
kind: Certificate
metadata:
  name: linkerd-identity-issuer
  namespace: linkerd
spec:
  secretName: linkerd-identity-issuer
  duration: 48h
  renewBefore: 25h
  issuerRef:
    name: linkerd-trust-anchor
    kind: Issuer
  commonName: identity.linkerd.cluster.local
  dnsNames:
  - identity.linkerd.cluster.local
  isCA: true
  privateKey:
    algorithm: ECDSA
  usages:
  - cert sign
  - crl sign
  - server auth
  - client auth
---
apiVersion: cert-manager.io/v1
kind: Certificate
metadata:
  name: linkerd-policy-validator
  namespace: linkerd
spec:
  secretName: linkerd-policy-validator-k8s-tls
  duration: 48h
  renewBefore: 25h
  issuerRef:
    name: webhook-issuer
    kind: Issuer
  commonName: linkerd-policy-validator.linkerd.svc
  dnsNames:
  - linkerd-policy-validator.linkerd.svc
  isCA: false
  privateKey:
    algorithm: ECDSA
    encoding: PKCS8
  usages:
  - server auth
---
apiVersion: cert-manager.io/v1
kind: Certificate
metadata:
  name: linkerd-proxy-injector
  namespace: linkerd
spec:
  secretName: linkerd-proxy-injector-k8s-tls
  duration: 48h
  renewBefore: 25h
  issuerRef:
    name: webhook-issuer
    kind: Issuer
  commonName: linkerd-proxy-injector.linkerd.svc
  dnsNames:
  - linkerd-proxy-injector.linkerd.svc
  isCA: false
  privateKey:
    algorithm: ECDSA
  usages:
  - server auth
---
apiVersion: cert-manager.io/v1
kind: Certificate
metadata:
  name: linkerd-sp-validator
  namespace: linkerd
spec:
  secretName: linkerd-sp-validator-k8s-tls
  duration: 48h
  renewBefore: 25h
  issuerRef:
    name: webhook-issuer
    kind: Issuer
  commonName: linkerd-sp-validator.linkerd.svc
  dnsNames:
  - linkerd-sp-validator.linkerd.svc
  isCA: false
  privateKey:
    algorithm: ECDSA
  usages:
  - server auth
---
# ignore if not using the viz extension
apiVersion: cert-manager.io/v1
kind: Certificate
metadata:
  name: tap
  namespace: linkerd-viz
spec:
  secretName: tap-k8s-tls
  duration: 48h
  renewBefore: 25h
  issuerRef:
    name: webhook-issuer
    kind: Issuer
  commonName: tap.linkerd-viz.svc
  dnsNames:
  - tap.linkerd-viz.svc
  isCA: false
  privateKey:
    algorithm: ECDSA
  usages:
  - server auth
---
# ignore if not using the viz extension
apiVersion: cert-manager.io/v1
kind: Certificate
metadata:
  name: linkerd-tap-injector
  namespace: linkerd-viz
spec:
  secretName: tap-injector-k8s-tls
  duration: 48h
  renewBefore: 25h
  issuerRef:
    name: webhook-issuer
    kind: Issuer
  commonName: tap-injector.linkerd-viz.svc
  dnsNames:
  - tap-injector.linkerd-viz.svc
  isCA: false
  privateKey:
    algorithm: ECDSA
  usages:
  - server auth
```

**Comment by adleong** [comment]
Hi @rootik.  Actually, the error message from check is a little bit misleading here.  That check will look for a secret named either `linkerd-tap-tls` or `tap-k8s-tls` and will only error if neither one is found.  I recommend double checking that the `tap-k8s-tls` secret exists as expected.

**Comment by rootik** [comment]
@adleong It seems it has been fixed in stable-2.12.2 (Helm 30.3.4)
Nothing has changed in our linkerd configuration except the versions but `linkerd viz check` doesn't complain about missing tap certificate secret anymore. Checked it on 3 clusters.
I think this issue can be closed.
